pkg/shares/reconstruct.go

```diff
@@ -0,0 +1,72 @@
+package shares
+
+import (
+	"bytes"
+
+	"github.com/celestiaorg/celestia-app/pkg/appconsts"
+	"github.com/celestiaorg/rsmt2d"
+	coretypes "github.com/tendermint/tendermint/types"
+)
+
+// reconstruct reconstructs block data from an extended data square.
+// TODO: export this function
+func reconstruct(eds *rsmt2d.ExtendedDataSquare) (coretypes.Data, error) {
```

Comment:
[not blocking]
I understand changing this name, but it might be a tad confusing since celestia-node using the term reconstruction for a very different thing. 

Also, it deviates from the existing splitting merging terminology. Not going to block on it though, so will defer to you on that.

Reply:
> Also, it deviates from the existing splitting merging terminology.

That was my intention because splitting and merging sound like inverse operations when they aren't
- **Split** converts block data into shares
- **Parse** converts shares into txs, blobs, or share sequences
- **Merge** or **Reconstruct** converts an extended data square into block data 

I can revert the rename if reviewers don't have alternative proposals

Reply:
Resolved by reverting the rename.